index.js

```diff
@@ -6,30 +6,20 @@
  *
  * `npm install typedarray-to-buffer`
  */
- /* eslint-disable no-proto */
 
 var isTypedArray = require('is-typedarray').strict
 
-module.exports = function (arr) {
-  // If `Buffer` is the browser `buffer` module, and the browser supports typed arrays,
-  // then avoid a copy. Otherwise, create a `Buffer` with a copy.
-  var constructor = Buffer.TYPED_ARRAY_SUPPORT
-    ? function (arr) {
-      arr.__proto__ = Buffer.prototype
-      return arr
+module.exports = function typedarrayToBuffer (arr) {
+  if (isTypedArray(arr)) {
+    // To avoid a copy, use the typed array's underlying ArrayBuffer to back new Buffer
+    var buf = new Buffer(arr.buffer)
+    if (arr.byteOffset !== 0 || arr.byteLength !== arr.buffer.byteLength) {
```

Comment:
The first check is actually superfluous, because if both lengths are equal, there can't be an offset.